File: lib/features/auth/auth_screen.dart
import 'package:flutter/material.dart';
import '../../core/config/app_config.dart';
import '../../core/constants/app_colors.dart';
import '../../core/utils/app_toast.dart';
import '../../data/repositories/organization_repository.dart';
import '../../data/repositories/user_profile_repository.dart';

/// Layar Masuk Utama (Auth Gate).
///
/// Pengguna wajib masuk sebelum dapat mengakses pembukuan dan riwayat kas.
class AuthScreen extends StatefulWidget {
  const AuthScreen({super.key, required this.config});

  final AppConfig config;

  @override
  State<AuthScreen> createState() => _AuthScreenState();
}

class _AuthScreenState extends State<AuthScreen> {
  final _userRepo = UserProfileRepository.instance;
  final _orgRepo = OrganizationRepository.instance;

  void _showGoogleAuthFlow(BuildContext context) {
    final emailCtrl = TextEditingController();
    final formKey = GlobalKey<FormState>();

    showModalBottomSheet<void>(
      context: context,
      isScrollControlled: true,
      shape: const RoundedRectangleBorder(
        borderRadius: BorderRadius.vertical(top: Radius.circular(24)),
      ),
      backgroundColor: Colors.white,
      builder: (ctx) {
        return Padding(
          padding: EdgeInsets.only(
            left: 20,
            right: 20,
            top: 24,
            bottom: MediaQuery.of(ctx).viewInsets.bottom + 24,
          ),
          child: Form(
            key: formKey,
            child: Column(
              mainAxisSize: MainAxisSize.min,
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: [
                    Row(
                      children: [
                        _buildGoogleIcon(size: 24),
                        const SizedBox(width: 10),
                        const Text(
                          'Masuk Akun Google',
                          style: TextStyle(
                            fontSize: 16,
                            fontWeight: FontWeight.bold,
                            color: AppColors.textPrimaryLight,
                          ),
                        ),
                      ],
                    ),
                    IconButton(
                      icon: const Icon(Icons.close, size: 20),
                      onPressed: () => Navigator.pop(ctx),
                    ),
                  ],
                ),
                const SizedBox(height: 6),
                const Text(
                  'Masukkan alamat email Google untuk verifikasi akun:',
                  style: TextStyle(
                    fontSize: 12,
                    color: AppColors.textSecondaryLight,
                  ),
                ),
                const SizedBox(height: 16),
                const Text(
                  'Email Google (@gmail.com)',
                  style: TextStyle(fontSize: 12, fontWeight: FontWeight.bold),
                ),
                const SizedBox(height: 6),
                TextFormField(
                  controller: emailCtrl,
                  keyboardType: TextInputType.emailAddress,
                  autofocus: true,
                  decoration: const InputDecoration(
                    hintText: 'nama.anda@gmail.com',
                    prefixIcon: Icon(Icons.email_outlined, size: 20),
                  ),
                  validator: (v) {
                    if (v == null || v.trim().isEmpty) {
                      return 'Email wajib diisi';
                    }
                    if (!v.contains('@') || !v.contains('.')) {
                      return 'Format email tidak valid';
                    }
                    return null;
                  },
                ),
                const SizedBox(height: 20),
                SizedBox(
                  width: double.infinity,
                  height: 48,
                  child: ElevatedButton(
                    onPressed: () async {
                      if (!formKey.currentState!.validate()) return;
                      final email = emailCtrl.text.trim();
                      Navigator.pop(ctx);

                      // Cek apakah akun sudah terdaftar
                      final existingUser = _userRepo.checkGoogleAccount(email);
                      if (existingUser != null) {
                        // USER LAMA -> Langsung login
                        await _userRepo.loginWithExistingGoogle(existingUser);
                        if (mounted) {
                          AppToast.success(
                            context,
                            'Selamat datang kembali, ${existingUser.name}',
                          );
                        }
                      } else {
                        // USER BARU -> Buka formulir nama & nomor WhatsApp
                        if (mounted) {
                          _showNewUserOnboarding(context, email);
                        }
                      }
                    },
                    style: ElevatedButton.styleFrom(
                      backgroundColor: AppColors.primaryRoyal,
                      foregroundColor: Colors.white,
                      shape: RoundedRectangleBorder(
                        borderRadius: BorderRadius.circular(14),
                      ),
                    ),
                    child: const Text(
                      'Lanjutkan',
                      style: TextStyle(fontWeight: FontWeight.bold),
                    ),
                  ),
                ),
              ],
            ),
          ),
        );
      },
    );
  }

  void _showNewUserOnboarding(BuildContext context, String googleEmail) {
    final nameCtrl = TextEditingController();
    final phoneCtrl = TextEditingController();
    final addressCtrl = TextEditingController();
    final formKey = GlobalKey<FormState>();

    showModalBottomSheet<void>(
      context: context,
      isScrollControlled: true,
      isDismissible: false,
      enableDrag: false,
      shape: const RoundedRectangleBorder(
        borderRadius: BorderRadius.vertical(top: Radius.circular(24)),
      ),
      backgroundColor: Colors.white,
      builder: (ctx) {
        return Padding(
          padding: EdgeInsets.only(
            left: 20,
            right: 20,
            top: 24,
            bottom: MediaQuery.of(ctx).viewInsets.bottom + 24,
          ),
          child: Form(
            key: formKey,
            child: SingleChildScrollView(
              child: Column(
                mainAxisSize: MainAxisSize.min,
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Row(
                    children: [
                      Container(
                        width: 4,
                        height: 18,
                        decoration: BoxDecoration(
                          color: AppColors.primaryRoyal,
                          borderRadius: BorderRadius.circular(2),
                        ),
                      ),
                      const SizedBox(width: 8),
                      const Text(
                        'Pendaftaran Warga Baru',
                        style: TextStyle(
                          fontSize: 16,
                          fontWeight: FontWeight.bold,
                          color: AppColors.textPrimaryLight,
                        ),
                      ),
                    ],
                  ),
                  const SizedBox(height: 6),
                  Text(
                    'Akun: $googleEmail',
                    style: const TextStyle(
                      fontSize: 12,
                      color: AppColors.primaryRoyal,
                      fontWeight: FontWeight.w600,
                    ),
                  ),
                  const SizedBox(height: 16),

                  // Nama Lengkap
                  const Text('Nama Lengkap',
                      style: TextStyle(fontSize: 12, fontWeight: FontWeight.bold)),
                  const SizedBox(height: 4),
                  TextFormField(
                    controller: nameCtrl,
                    decoration: const InputDecoration(
                      hintText: 'Nama lengkap sesuai KTP',
                      prefixIcon: Icon(Icons.person_outline, size: 20),
                    ),
                    validator: (v) =>
                        (v == null || v.trim().isEmpty) ? 'Nama wajib diisi' : null,
                  ),
                  const SizedBox(height: 14),

                  // Nomor WhatsApp
                  const Text('Nomor WhatsApp',
                      style: TextStyle(fontSize: 12, fontWeight: FontWeight.bold)),
                  const SizedBox(height: 4),
                  TextFormField(
                    controller: phoneCtrl,
                    keyboardType: TextInputType.phone,
                    decoration: const InputDecoration(
                      hintText: 'Contoh: 0812-3456-7890',
                      prefixIcon: Icon(Icons.phone_outlined, size: 20),
                    ),
                    validator: (v) =>
                        (v == null || v.trim().isEmpty) ? 'Nomor WhatsApp wajib diisi' : null,
                  ),
                  const SizedBox(height: 14),

                  // Alamat Rumah
                  const Text('Alamat & RT/RW',
                      style: TextStyle(fontSize: 12, fontWeight: FontWeight.bold)),
                  const SizedBox(height: 4),
                  TextFormField(
                    controller: addressCtrl,
                    decoration: const InputDecoration(
                      hintText: 'Contoh: RT 02 / RW 05',
                      prefixIcon: Icon(Icons.location_on_outlined, size: 20),
                    ),
                    validator: (v) =>
                        (v == null || v.trim().isEmpty) ? 'Alamat wajib diisi' : null,
                  ),
                  const SizedBox(height: 22),

                  // Tombol Selesai
                  SizedBox(
                    width: double.infinity,
                    height: 48,
                    child: ElevatedButton(
                      onPressed: () async {
                        if (!formKey.currentState!.validate()) return;
                        final name = nameCtrl.text.trim();
                        final phone = phoneCtrl.text.trim();
                        final address = addressCtrl.text.trim();

                        final err = await _userRepo.registerWithGoogle(
                          email: googleEmail,
                          name: name,
                          phone: phone,
                          address: address,
                        );

                        if (!context.mounted) return;
                        if (err != null) {
                          AppToast.error(context, err);
                        } else {
                          Navigator.pop(ctx);
                          AppToast.success(
                            context,
                            'Pendaftaran berhasil. Selamat datang, $name!',
                          );
                        }
                      },
                      style: ElevatedButton.styleFrom(
                        backgroundColor: AppColors.incomeGreen,
                        foregroundColor: Colors.white,
                        shape: RoundedRectangleBorder(
                          borderRadius: BorderRadius.circular(14),
                        ),
                      ),
                      child: const Text(
                        'Daftar & Masuk',
                        style: TextStyle(
                          fontSize: 14,
                          fontWeight: FontWeight.bold,
                        ),
                      ),
                    ),
                  ),
                ],
              ),
            ),
          ),
        );
      },
    );
  }

  void _showAdminPinModal(BuildContext context) {
    final pinCtrl = TextEditingController();
    showDialog<void>(
      context: context,
      builder: (ctx) => AlertDialog(
        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(20)),
        title: const Text(
          'Akses Pengurus',
          style: TextStyle(fontWeight: FontWeight.bold, fontSize: 16),
        ),
        content: Column(
          mainAxisSize: MainAxisSize.min,
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const Text(
              'Masukkan PIN akses pengurus untuk mengelola kas organisasi:',
              style: TextStyle(fontSize: 12, color: AppColors.textSecondaryLight),
            ),
            const SizedBox(height: 12),
            TextField(
              controller: pinCtrl,
              keyboardType: TextInputType.number,
              obscureText: true,
              decoration: const InputDecoration(
                hintText: 'PIN Pengurus (Default: 123456)',
                prefixIcon: Icon(Icons.lock_outline),
              ),
            ),
          ],
        ),
        actions: [
          TextButton(
            onPressed: () => Navigator.pop(ctx),
            child: const Text('Batal'),
          ),
          FilledButton(
            onPressed: () async {
              final pin = pinCtrl.text.trim();
              final success = await _userRepo.loginWithAdminPin(pin);
              if (!ctx.mounted) return;
              Navigator.pop(ctx);
              if (success) {
                AppToast.success(context, 'Masuk sebagai Bendahara Kas');
              } else {
                AppToast.error(context, 'PIN pengurus tidak valid');
              }
            },
            child: const Text('Masuk'),
          ),
        ],
      ),
    );
  }

  Widget _buildGoogleIcon({double size = 20}) {
    return Container(
      width: size,
      height: size,
      decoration: BoxDecoration(
        color: Colors.white,
        shape: BoxShape.circle,
        border: Border.all(color: AppColors.borderSubtle, width: 0.8),
      ),
      child: Center(
        child: Text(
          'G',
          style: TextStyle(
            color: const Color(0xFF4285F4),
            fontWeight: FontWeight.w900,
            fontSize: size * 0.65,
          ),
        ),
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColors.backgroundLight,
      body: SafeArea(
        child: Center(
          child: SingleChildScrollView(
            padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 32),
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                // Logo Kas Go
                Container(
                  width: 80,
                  height: 80,
                  decoration: BoxDecoration(
                    gradient: const LinearGradient(
                      colors: [
                        AppColors.heroPurpleStart,
                        AppColors.heroPurpleEnd,
                      ],
                      begin: Alignment.topLeft,
                      end: Alignment.bottomRight,
                    ),
                    shape: BoxShape.circle,
                    boxShadow: [
                      BoxShadow(
                        color: AppColors.primaryRoyal.withOpacity(0.25),
                        blurRadius: 20,
                        offset: const Offset(0, 8),
                      ),
                    ],
                  ),
                  child: const Center(
                    child: Icon(
                      Icons.account_balance_wallet,
                      color: AppColors.accentGold,
                      size: 38,
                    ),
                  ),
                ),
                const SizedBox(height: 18),

                // Judul Aplikasi
                const Text(
                  'Kas Go',
                  style: TextStyle(
                    fontSize: 28,
                    fontWeight: FontWeight.w900,
                    letterSpacing: -0.5,
                    color: AppColors.textPrimaryLight,
                  ),
                ),
                const SizedBox(height: 4),
                AnimatedBuilder(
                  animation: _orgRepo,
                  builder: (context, _) {
                    return Text(
                      _orgRepo.organizationTitle.isNotEmpty
                          ? _orgRepo.organizationTitle
                          : 'Aplikasi Kas Karang Taruna',
                      style: const TextStyle(
                        fontSize: 13,
                        fontWeight: FontWeight.w500,
                        color: AppColors.textSecondaryLight,
                      ),
                      textAlign: TextAlign.center,
                    );
                  },
                ),
                const SizedBox(height: 36),

                // Card Utama Masuk
                Container(
                  width: double.infinity,
                  padding: const EdgeInsets.all(24),
                  decoration: BoxDecoration(
                    color: Colors.white,
                    borderRadius: BorderRadius.circular(24),
                    border: Border.all(color: AppColors.borderSubtle),
                    boxShadow: [
                      BoxShadow(
                        color: Colors.black.withOpacity(0.03),
                        blurRadius: 16,
                        offset: const Offset(0, 6),
                      ),
                    ],
                  ),
                  child: Column(
                    children: [
                      const Text(
                        'Silakan masuk untuk melanjutkan',
                        style: TextStyle(
                          fontSize: 13,
                          fontWeight: FontWeight.w600,
                          color: AppColors.textSecondaryLight,
                        ),
                      ),
                      const SizedBox(height: 20),

                      // Tombol Masuk dengan Google Nyata
                      SizedBox(
                        width: double.infinity,
                        height: 52,
                        child: OutlinedButton(
                          onPressed: () => _showGoogleAuthFlow(context),
                          style: OutlinedButton.styleFrom(
                            side: const BorderSide(
                              color: AppColors.primaryRoyal,
                              width: 1.5,
                            ),
                            backgroundColor: Colors.white,
                            shape: RoundedRectangleBorder(
                              borderRadius: BorderRadius.circular(16),
                            ),
                          ),
                          child: Row(
                            mainAxisAlignment: MainAxisAlignment.center,
                            children: [
                              _buildGoogleIcon(size: 24),
                              const SizedBox(width: 12),
                              const Text(
                                'Masuk dengan Google',
                                style: TextStyle(
                                  fontSize: 14,
                                  fontWeight: FontWeight.bold,
                                  color: AppColors.primaryRoyal,
                                ),
                              ),
                            ],
                          ),
                        ),
                      ),
                      const SizedBox(height: 18),

                      // Pembatas
                      Row(
                        children: [
                          const Expanded(child: Divider(color: Color(0xFFF1F5F9))),
                          Padding(
                            padding: const EdgeInsets.symmetric(horizontal: 10),
                            child: Text(
                              'atau',
                              style: TextStyle(
                                fontSize: 11,
                                color: Colors.grey.shade400,
                              ),
                            ),
                          ),
                          const Expanded(child: Divider(color: Color(0xFFF1F5F9))),
                        ],
                      ),
                      const SizedBox(height: 14),

                      // Tombol Masuk sebagai Pengurus
                      TextButton.icon(
                        onPressed: () => _showAdminPinModal(context),
                        icon: const Icon(
                          Icons.admin_panel_settings_outlined,
                          size: 18,
                          color: AppColors.primaryRoyal,
                        ),
                        label: const Text(
                          'Masuk sebagai Pengurus',
                          style: TextStyle(
                            fontSize: 13,
                            fontWeight: FontWeight.bold,
                            color: AppColors.primaryRoyal,
                          ),
                        ),
                      ),
                    ],
                  ),
                ),
                const SizedBox(height: 32),

                // Footer
                const Text(
                  'Kas Go • Versi 1.0.5',
                  style: TextStyle(
                    fontSize: 11,
                    color: AppColors.textMutedLight,
                  ),
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}

File: app/src/main/java/id/or/karangtaruna/kasgo/ui/screens/auth/AuthScreen.kt
package id.or.karangtaruna.kasgo.ui.screens.auth

import android.app.Activity
import androidx.activity.compose.rememberLauncherForActivityResult
import androidx.activity.result.contract.ActivityResultContracts
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.AdminPanelSettings
import androidx.compose.material.icons.outlined.Email
import androidx.compose.material.icons.outlined.LocationOn
import androidx.compose.material.icons.outlined.Lock
import androidx.compose.material.icons.outlined.Person
import androidx.compose.material.icons.outlined.Phone
import androidx.compose.material.icons.rounded.AccountBalanceWallet
import androidx.compose.material.icons.rounded.Close
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Divider
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.ModalBottomSheet
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.material3.rememberModalBottomSheetState
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.google.android.gms.auth.api.signin.GoogleSignIn
import com.google.android.gms.auth.api.signin.GoogleSignInOptions
import com.google.android.gms.common.api.ApiException
import id.or.karangtaruna.kasgo.core.constants.AppColors
import id.or.karangtaruna.kasgo.core.utils.AppToast
import id.or.karangtaruna.kasgo.data.repositories.OrganizationRepository
import id.or.karangtaruna.kasgo.data.repositories.UserProfileRepository

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun AuthScreen() {
    val context = LocalContext.current
    val userRepo = remember { UserProfileRepository.get() }
    val orgRepo = remember { OrganizationRepository.get() }
    val orgConfig by orgRepo.config.collectAsState()

    var showNewUserModal by remember { mutableStateOf(false) }
    var newUserEmail by remember { mutableStateOf("") }
    var newUserName by remember { mutableStateOf("") }

    var showAdminPinDialog by remember { mutableStateOf(false) }
    var adminPinText by remember { mutableStateOf("") }

    var showManualEmailSheet by remember { mutableStateOf(false) }
    var manualEmailText by remember { mutableStateOf("") }

    // Google Sign-In Client for native Android Account Chooser popup
    val gso = remember {
        GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
            .requestEmail()
            .build()
    }
    val googleSignInClient = remember { GoogleSignIn.getClient(context, gso) }

    fun processEmailAuth(email: String, displayName: String) {
        val cleanEmail = email.trim().lowercase()
        val existing = userRepo.checkGoogleAccount(cleanEmail)
        if (existing != null) {
            userRepo.loginWithExistingGoogle(existing)
            AppToast.success("Selamat datang kembali, ${existing.name}")
        } else {
            newUserEmail = cleanEmail
            newUserName = displayName.ifBlank { "Warga Baru" }
            showNewUserModal = true
        }
    }

    val googleSignInLauncher = rememberLauncherForActivityResult(
        contract = ActivityResultContracts.StartActivityForResult()
    ) { result ->
        if (result.resultCode == Activity.RESULT_OK) {
            val task = GoogleSignIn.getSignedInAccountFromIntent(result.data)
            try {
                val account = task.getResult(ApiException::class.java)
                val email = account?.email
                val name = account?.displayName ?: ""
                if (!email.isNullOrBlank()) {
                    processEmailAuth(email, name)
                } else {
                    AppToast.error("Tidak dapat memperoleh email dari akun Google")
                }
            } catch (e: ApiException) {
                // If Play Services OAuth isn't configured in console, allow manual email input
                showManualEmailSheet = true
            }
        } else {
            // Prompt fallback sheet if user cancelled or error
            showManualEmailSheet = true
        }
    }

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(AppColors.backgroundLight),
        contentAlignment = Alignment.Center
    ) {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .verticalScroll(rememberScrollState())
                .padding(horizontal = 24.dp, vertical = 36.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Spacer(modifier = Modifier.height(32.dp))

            // Logo Monogram
            Box(
                modifier = Modifier
                    .size(80.dp)
                    .shadow(16.dp, CircleShape, spotColor = AppColors.primaryRoyal)
                    .clip(CircleShape)
                    .background(
                        Brush.linearGradient(
                            listOf(AppColors.heroPurpleStart, AppColors.heroPurpleEnd)
                        )
                    ),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    imageVector = Icons.rounded.AccountBalanceWallet,
                    contentDescription = null,
                    tint = AppColors.accentGold,
                    modifier = Modifier.size(38.dp)
                )
            }
            Spacer(modifier = Modifier.height(16.dp))

            Text(
                text = "Kas Go",
                fontSize = 28.sp,
                fontWeight = FontWeight.ExtraBold,
                color = AppColors.textPrimaryLight,
                letterSpacing = (-0.5).sp
            )
            Spacer(modifier = Modifier.height(4.dp))
            Text(
                text = orgConfig.fullTitle,
                fontSize = 13.sp,
                color = AppColors.textSecondaryLight,
                fontWeight = FontWeight.Medium,
                textAlign = TextAlign.Center
            )

            Spacer(modifier = Modifier.height(36.dp))

            // Card Masuk
            Surface(
                modifier = Modifier
                    .fillMaxWidth()
                    .shadow(14.dp, RoundedCornerShape(24.dp)),
                shape = RoundedCornerShape(24.dp),
                color = Color.White
            ) {
                Column(
                    modifier = Modifier.padding(24.dp),
                    horizontalAlignment = Alignment.CenterHorizontally
                ) {
                    Text(
                        text = "Silakan masuk untuk melanjutkan",
                        fontSize = 13.sp,
                        fontWeight = FontWeight.SemiBold,
                        color = AppColors.textSecondaryLight
                    )
                    Spacer(modifier = Modifier.height(20.dp))

                    // Tombol Masuk dengan Google NATIVE
                    OutlinedButton(
                        onClick = {
                            try {
                                googleSignInClient.signOut() // Clear cache to always show chooser
                                googleSignInLauncher.launch(googleSignInClient.signInIntent)
                            } catch (_: Exception) {
                                showManualEmailSheet = true
                            }
                        },
                        modifier = Modifier
                            .fillMaxWidth()
                            .height(52.dp),
                        shape = RoundedCornerShape(16.dp),
                        colors = ButtonDefaults.outlinedButtonColors(
                            containerColor = Color.White
                        ),
                        border = androidx.compose.foundation.BorderStroke(
                            1.5.dp,
                            AppColors.primaryRoyal
                        )
                    ) {
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Box(
                                modifier = Modifier
                                    .size(24.dp)
                                    .clip(CircleShape)
                                    .border(1.dp, AppColors.borderSubtle, CircleShape)
                                    .background(Color.White),
                                contentAlignment = Alignment.Center
                            ) {
                                Text(
                                    text = "G",
                                    fontWeight = FontWeight.Black,
                                    fontSize = 15.sp,
                                    color = Color(0xFF4285F4)
                                )
                            }
                            Spacer(modifier = Modifier.width(12.dp))
                            Text(
                                text = "Masuk dengan Google",
                                fontSize = 14.sp,
                                fontWeight = FontWeight.Bold,
                                color = AppColors.primaryRoyal
                            )
                        }
                    }

                    Spacer(modifier = Modifier.height(18.dp))

                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Divider(modifier = Modifier.weight(1f), color = Color(0xFFF1F5F9))
                        Text(
                            text = "atau",
                            fontSize = 11.sp,
                            color = AppColors.textMutedLight,
                            modifier = Modifier.padding(horizontal = 10.dp)
                        )
                        Divider(modifier = Modifier.weight(1f), color = Color(0xFFF1F5F9))
                    }

                    Spacer(modifier = Modifier.height(14.dp))

                    // Masuk sebagai Pengurus
                    TextButton(
                        onClick = { showAdminPinDialog = true }
                    ) {
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Icon(
                                imageVector = Icons.outlined.AdminPanelSettings,
                                contentDescription = null,
                                tint = AppColors.primaryRoyal,
                                modifier = Modifier.size(18.dp)
                            )
                            Spacer(modifier = Modifier.width(6.dp))
                            Text(
                                text = "Masuk sebagai Pengurus",
                                fontSize = 13.sp,
                                fontWeight = FontWeight.Bold,
                                color = AppColors.primaryRoyal
                            )
                        }
                    }
                }
            }

            Spacer(modifier = Modifier.height(32.dp))

            Text(
                text = "Kas Go • Versi ${id.or.karangtaruna.kasgo.services.AppUpdateService.currentVersion}",
                fontSize = 11.sp,
                color = AppColors.textMutedLight
            )
        }
    }

    // Modal Sheet 1: Manual Google Email Fallback (jika perangkat tidak ada Play Services)
    if (showManualEmailSheet) {
        ModalBottomSheet(
            onDismissRequest = { showManualEmailSheet = false },
            sheetState = rememberModalBottomSheetState(),
            containerColor = Color.White
        ) {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 24.dp, vertical = 20.dp)
            ) {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Box(
                        modifier = Modifier
                            .size(24.dp)
                            .clip(CircleShape)
                            .background(Color.White)
                            .border(1.dp, AppColors.borderSubtle, CircleShape),
                        contentAlignment = Alignment.Center
                    ) {
                        Text(
                            text = "G",
                            fontWeight = FontWeight.Black,
                            fontSize = 14.sp,
                            color = Color(0xFF4285F4)
                        )
                    }
                    Spacer(modifier = Modifier.width(10.dp))
                    Text(
                        text = "Pilih / Masukkan Akun Google",
                        fontSize = 16.sp,
                        fontWeight = FontWeight.Bold,
                        color = AppColors.textPrimaryLight
                    )
                }
                Spacer(modifier = Modifier.height(16.dp))
                OutlinedTextField(
                    value = manualEmailText,
                    onValueChange = { manualEmailText = it },
                    label = { Text("Email Google") },
                    placeholder = { Text("nama.anda@gmail.com") },
                    leadingIcon = { Icon(Icons.outlined.Email, null) },
                    singleLine = true,
                    keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Email),
                    modifier = Modifier.fillMaxWidth()
                )
                Spacer(modifier = Modifier.height(20.dp))
                Button(
                    onClick = {
                        val email = manualEmailText.trim()
                        if (email.isNotBlank() && email.contains("@")) {
                            showManualEmailSheet = false
                            processEmailAuth(email, email.substringBefore("@"))
                        } else {
                            AppToast.error("Masukkan alamat email yang valid")
                        }
                    },
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(48.dp),
                    shape = RoundedCornerShape(14.dp),
                    colors = ButtonDefaults.buttonColors(containerColor = AppColors.primaryRoyal)
                ) {
                    Text("Lanjutkan", fontWeight = FontWeight.Bold)
                }
                Spacer(modifier = Modifier.height(16.dp))
            }
        }
    }

    // Modal Sheet 2: Formulir Pendaftaran Warga Baru (Nama Lengkap, Nomor WhatsApp, Alamat)
    if (showNewUserModal) {
        var inputName by remember { mutableStateOf(newUserName) }
        var inputPhone by remember { mutableStateOf("") }
        var inputAddress by remember { mutableStateOf("") }

        ModalBottomSheet(
            onDismissRequest = { /* Must complete registration */ },
            sheetState = rememberModalBottomSheetState(),
            containerColor = Color.White
        ) {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .verticalScroll(rememberScrollState())
                    .padding(horizontal = 24.dp, vertical = 20.dp)
            ) {
                Text(
                    text = "Pendaftaran Warga Baru",
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Bold,
                    color = AppColors.textPrimaryLight
                )
                Spacer(modifier = Modifier.height(4.dp))
                Text(
                    text = "Akun Google: $newUserEmail",
                    fontSize = 12.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = AppColors.primaryRoyal
                )
                Spacer(modifier = Modifier.height(16.dp))

                OutlinedTextField(
                    value = inputName,
                    onValueChange = { inputName = it },
                    label = { Text("Nama Lengkap") },
                    placeholder = { Text("Sesuai KTP") },
                    leadingIcon = { Icon(Icons.outlined.Person, null) },
                    singleLine = true,
                    modifier = Modifier.fillMaxWidth()
                )
                Spacer(modifier = Modifier.height(12.dp))

                OutlinedTextField(
                    value = inputPhone,
                    onValueChange = { inputPhone = it },
                    label = { Text("Nomor WhatsApp") },
                    placeholder = { Text("0812-xxxx-xxxx") },
                    leadingIcon = { Icon(Icons.outlined.Phone, null) },
                    singleLine = true,
                    keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Phone),
                    modifier = Modifier.fillMaxWidth()
                )
                Spacer(modifier = Modifier.height(12.dp))

                OutlinedTextField(
                    value = inputAddress,
                    onValueChange = { inputAddress = it },
                    label = { Text("Alamat Rumah & RT/RW") },
                    placeholder = { Text("Contoh: RT 02 / RW 05") },
                    leadingIcon = { Icon(Icons.outlined.LocationOn, null) },
                    singleLine = true,
                    modifier = Modifier.fillMaxWidth()
                )
                Spacer(modifier = Modifier.height(24.dp))

                Button(
                    onClick = {
                        val name = inputName.trim()
                        val phone = inputPhone.trim()
                        val address = inputAddress.trim()

                        if (name.isBlank()) {
                            AppToast.error("Nama lengkap wajib diisi")
                            return@Button
                        }
                        if (phone.isBlank()) {
                            AppToast.error("Nomor WhatsApp wajib diisi")
                            return@Button
                        }
                        if (address.isBlank()) {
                            AppToast.error("Alamat rumah wajib diisi")
                            return@Button
                        }

                        val err = userRepo.registerWithGoogle(
                            email = newUserEmail,
                            name = name,
                            phone = phone,
                            address = address
                        )
                        if (err != null) {
                            AppToast.error(err)
                        } else {
                            showNewUserModal = false
                            AppToast.success("Pendaftaran berhasil. Selamat datang, $name!")
                        }
                    },
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(48.dp),
                    shape = RoundedCornerShape(14.dp),
                    colors = ButtonDefaults.buttonColors(containerColor = AppColors.incomeGreen)
                ) {
                    Text("Daftar & Masuk", fontWeight = FontWeight.Bold, fontSize = 14.sp)
                }
                Spacer(modifier = Modifier.height(16.dp))
            }
        }
    }

    // Dialog Akses Pengurus (PIN)
    if (showAdminPinDialog) {
        AlertDialog(
            onDismissRequest = { showAdminPinDialog = false },
            title = {
                Text(
                    text = "Akses Pengurus",
                    fontWeight = FontWeight.Bold,
                    fontSize = 16.sp
                )
            },
            text = {
                Column {
                    Text(
                        text = "Masukkan PIN akses pengurus untuk mengelola kas:",
                        fontSize = 12.sp,
                        color = AppColors.textSecondaryLight
                    )
                    Spacer(modifier = Modifier.height(12.dp))
                    OutlinedTextField(
                        value = adminPinText,
                        onValueChange = { adminPinText = it },
                        placeholder = { Text("PIN Pengurus (Default: 123456)") },
                        leadingIcon = { Icon(Icons.outlined.Lock, null) },
                        visualTransformation = PasswordVisualTransformation(),
                        keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.NumberPassword),
                        singleLine = true,
                        modifier = Modifier.fillMaxWidth()
                    )
                }
            },
            confirmButton = {
                Button(
                    onClick = {
                        val pin = adminPinText.trim()
                        val success = userRepo.loginWithAdminPin(pin)
                        if (success) {
                            showAdminPinDialog = false
                            AppToast.success("Masuk sebagai Bendahara Kas")
                        } else {
                            AppToast.error("PIN pengurus salah")
                        }
                    },
                    colors = ButtonDefaults.buttonColors(containerColor = AppColors.primaryRoyal)
                ) {
                    Text("Masuk")
                }
            },
            dismissButton = {
                TextButton(onClick = { showAdminPinDialog = false }) {
                    Text("Batal")
                }
            },
            containerColor = Color.White,
            shape = RoundedCornerShape(20.dp)
        )
    }
}
